src/hooks/useWrapUnwrapEth.ts

```diff
@@ -0,0 +1,89 @@
+import { Receipt, TxOptionalParams } from 'types'
+import { assert } from '@gnosis.pm/dex-js'
+import { wethApi } from 'api'
+import { useWalletConnection } from './useWalletConnection'
+import useSafeState from './useSafeState'
+import { WrapUnwrapParams } from 'api/weth/WethApi'
+import { logDebug } from 'utils'
+import { DISABLE_SPINNER_DELAY } from 'const'
+
+export interface Result {
+  wrappingEth: boolean
+  unwrappingWeth: boolean
+  wrapEth(amount: string, txOptionalParams?: TxOptionalParams): Promise<Receipt>
+  unwrapWeth(amount: string, txOptionalParams?: TxOptionalParams): Promise<Receipt>
+}
+
+interface WrapUnwrapAuxParams {
+  userAddress: string | undefined
+  networkId: number | undefined
+  isConnected: boolean | undefined
```

Comment:
nitpick: just my personal preference, ignore if you prefer your version

```suggestion
  userAddress?: string
  networkId?: number
  isConnected?: boolean
```

Reply:
always better this way :+1:

Reply:
Agree, changing...